code/modules/atmospherics/gasmixtures/gas_types.dm

```diff
@@ -40,32 +68,99 @@ GLOBAL_LIST_INIT(nonreactive_gases, typecacheof(list(/datum/gas/oxygen, /datum/g
+	/// Relative rarity compared to other gases, used when setting up the reactions list.
+	var/rarity = 0
+
+	// Condensation:
+	/// The maximum rate of condensation of this gas (as a function of total moles). If this is falsey the gas will not condense
+	var/cond_rate
+	/// The maximum temperature this gas will condense at
+	var/cond_temp_max
+	/// The temperature at and below which the gas will condense at it's maximum rate
+	var/cond_temp_min
+	/// The typepath of the reagent this gas condenses into. If this is null the reagents will not produce a gas when condensing
+	var/datum/reagent/cond_type
+	/// The amount of thermal energy released by one mole of this gas condensing
+	var/cond_heat = 0
+	/// A reference to the condensation event [CALLBACK] If this is truthy on init it will be repaced with the appropriate callback. Leave this as null
+	var/datum/callback/cond_event = null
+
+/datum/gas/New()
+	. = ..()
+	if(cond_event)
+		cond_event = CALLBACK(src, .proc/on_condense)
+
+/**
+ * A proc used to generate the condensation event callback
+ *
+ * If the callback is generated this will be called whenever the gas condenses in a condenser
+ *
+ * Arguments:
+ * - [holder][/datum/gas_mixture]: The gas mixture this is being condensed from
+ * - cond_rate: The rate at which this is condensing in moles
+ * - cond_temp: the temperature at which this is condensing in kelvin
+ * - [location][/atom]: The atom this is condensing on or in
+ * - [target][/datum/reagents]: The reagent mixture this gas is condensing to
+ * - [cond_rates][/list]: The list of all condensation rates in the current condensation operation
+ *   - Exists entirely in case someone wants to add a condensation effect that varies on simultaneous condensing gases
+ *   - For the sake of avoiding race conditions this should be treated as read-only
+ *   - NOTE: This does not actually contain the list of absolute condensation rates, only the list of relative condensation rates
+ * - rate_multiplier: The multiplier to apply to the condensation rates
+ * - delta_time: The amount of time the condensation is occuring over
+ */
+/datum/gas/proc/on_condense(datum/gas_mixture/holder, cond_moles, cond_temp, atom/location, datum/reagents/target, list/cond_rates, rate_multiplier, delta_time)
+	return
+
 
 /datum/gas/oxygen
 	id = "o2"
 	specific_heat = 20
 	name = "Oxygen"
 	rarity = 900
+	cond_rate = 0.1
+	cond_temp_max = 90
+	cond_temp_min = TCMB
+	cond_type = /datum/reagent/oxygen
+	cond_heat = 6828
 
 /datum/gas/nitrogen
 	id = "n2"
 	specific_heat = 20
 	name = "Nitrogen"
 	rarity = 1000
+	cond_rate = 0.1
+	cond_temp_max = 77
+	cond_temp_min = TCMB
+	cond_type = /datum/reagent/nitrogen
+	cond_heat = 5590
 
 /datum/gas/carbon_dioxide //what the fuck is this?
 	id = "co2"
 	specific_heat = 30
 	name = "Carbon Dioxide"
 	rarity = 700
+	cond_rate = 0.05
+	cond_temp_max = BP_CARBON_DIOXIDE
+	cond_temp_min = BP_CARBON_DIOXIDE
+	cond_type = /datum/reagent/carbondioxide
+	cond_heat = 25230
```

Comment:
Why did you use the numbers you did here/in the other gastypes?

Reply:
The condensation rate was pretty much arbitrary. The max condensation temp is just the temperature it boils at. The minimum condensation temp was again, pretty much arbitrary. I think I chose it to be equal to the condensation temp because CO2 turns into dry ice? I don't remember. I just googled the IRL enthalpy of vaporization for the condensation heat.